theme/static/js/resourceLandingAjax.js

```diff
@@ -470,4 +470,176 @@ function get_user_info_ajax_submit(url, obj) {
 
         }
     });
+}
+
+function delete_folder_ajax_submit(res_id, folder_path) {
+    $(".file-browser-container, #fb-files-container").css("cursor", "progress");
+
+    return $.ajax({
+        type: "POST",
+        url: '/hsapi/_internal/data-store-delete-folder/',
+        async: true,
+        data: {
+            res_id: res_id,
+            folder_path: folder_path
+        },
+        success: function (result) {
+        },
+        error: function(xhr, errmsg, err){
+        }
+    });
+}
+
+// This method is called to refresh the loader with the most recent structure after every other call
+function get_irods_folder_struct_ajax_submit(res_id, store_path) {
+    $("#fb-files-container, #fb-files-container").css("cursor", "progress");
+    return $.ajax({
+        type: "POST",
+        url: '/hsapi/_internal/data-store-structure/',
+        async: true,
+        data: {
+            res_id: res_id,
+            store_path: store_path
+        },
+        success: function (result) {
+            var files = result.files;
+            var folders = result.folders;
+            $('#fb-files-container').empty();
+            if (files.length > 0) {
+                $.each(files, function(i, v) {
+                    $('#fb-files-container').append(getFileTemplateInstance(v['name'], v['type'], v['size'], v['pk'], v['url']));
+                });
+            }
+            if (folders.length > 0) {
+                $.each(folders, function(i, v) {
+                    $('#fb-files-container').append(getFolderTemplateInstance(v));
+                });
+            }
+            if (!files.length && !folders.length) {
+                $('#fb-files-container').append('<span class="text-muted">This directory is empty</span>');
+            }
+
+            onSort();
+
+            bindFileBrowserItemEvents();
+
+            $("#hs-file-browser").attr("data-current-path", store_path);
+            $("#hs-file-browser").attr("data-res-id", res_id);
+
+            // strip the 'data' folder from the path
+            setBreadCrumbs(store_path.replace("data/", ""));
+
+            if ($("#fbContainmentWrapper").hasClass("dropzone")) {
+                // If no multiple files allowed and a file already exists, disable upload
+                var allowMultiple = $("#hs-file-browser").attr("data-allow-multiple-files") == "True";
+                if (!allowMultiple && files.length > 0) {
+                    $('.dz-input').hide();
+                    $(".fb-upload-caption").toggleClass("hidden", true);
+                }
+                else {
+
```

Comment:
@mauriel133 better to remove this extra line if you have other things to change so as to commit this together with other changes needed.